collector/application/application.go

```diff
@@ -58,14 +60,18 @@ func (a *Application) Run() error {
 	}
 	// Wait until the receiver shutdowns
 	err = a.receiver.Start()
+	a.shutdownWG.Add(1)
 	if err != nil {
 		return fmt.Errorf("failed to start application: %v", err)
 	}
+	a.shutdownWG.Wait()
 	return nil
 }
 
 func (a *Application) Shutdown() error {
-	return multierr.Combine(a.receiver.Shutdown(), a.analyzerManager.ShutdownAll(a.telemetry.Telemetry.Logger))
+	err := a.receiver.Shutdown()
+	a.shutdownWG.Done()
+	return multierr.Combine(err, a.analyzerManager.ShutdownAll(a.telemetry.Telemetry.Logger))
```

Comment:
`a.analyzerManager.ShutdownAll` may not execute normally since the main thread has already exited after `a.shutdownWG.Done()`.